PythonTool/TagExpansion.py

```diff
@@ -0,0 +1,140 @@
+    parser.add_argument("meshfile",
+                        type=str,
+                        help="Provide a path to the mesh file."
+                        )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def tag_expansion(mesh_file):
+    """
+    Expand the vector tags in the given mesh data file. Write a file to disk
+    for each time state with the corresponding scalar tag values from each vector tag.
+
+    Input:
+    ______
+       mesh_file: str
+           User supplied mesh file location.
+
+    Returns:
+    ________
+       none
+    """
+
+    # Load the mesh file to create a reference mesh.
+    mb2 = core.Core()
+    mb2.load_file(mesh_file)
+
+    # Retrieve an arbitrary MBHEX element in the mesh and extract the tag list.
+    root2 = mb2.get_root_set()
+    hexes2 = mb2.get_entities_by_type(root2, types.MBHEX)
+
+    if len(hexes2) == 0:
+        print("WARNING: No hex elements were found in this data file.")
+        exit()
+
+    tag_list2 = mb2.tag_get_tags_on_entity(hexes2[0])
+
+    # Check if each tag is a vector. If so, delete it.
+    for tag in tag_list2:
+        tag_length = mb2.tag_get_length(tag)
+        if tag_length > 1:
+            # Grab the length and name of the vector tag for later use.
+            reference_length = tag_length
+            tag_name = tag.get_name()
+            mb2.tag_delete(tag)
+
+    # Load the mesh file for extracting vector tag data.
+    mb1 = core.Core()
+    mb1.load_file(mesh_file)
+
+    # Retrieve an arbitrary MBHEX element in the mesh and extract the tag list.
+    root1 = mb1.get_root_set()
+    hexes1 = mb1.get_entities_by_type(root1, types.MBHEX)
+    tag_list1 = mb1.tag_get_tags_on_entity(hexes1[0])
+
+    # Ensure each vector tag is the same length.
+    vector_tags = []
+    for tag in tag_list1:
+        tag_length = mb1.tag_get_length(tag)
+        if tag_length > 1:
+            vector_tags.append(tag)
+            if mb1.tag_get_length(tag) is not reference_length:
+                print("WARNING: Found a vector tag of different length.")
+                exit()
+
+    # Make sure the mesh file contains at least one vector tag.
+    if len(vector_tags) < 1:
+        print("WARNING: This mesh file did not contain any vector tags.")
+        exit()
+
+    # Create a directory for the vector tag expansion files.
+    input_list = mesh_file.split("/")
+    file_name = str(input_list[-1]).split(".")
+    dir_name = file_name[0] + "_database"
+
+    # Ensure an existing dictionary is not written over.
+    dict_number = 1
+    while os.path.isdir(dir_name):
+        dir_name = file_name[0] + "_database" + str(dict_number)
+        dict_number += 1
+    os.mkdir(dir_name)
+
+    """
+    For each vector tag, retrieve the scalar value at a specific time state and
+    create a scalar tag on the reference mesh. For each time state, write the
+    scalar tag values to disk in a vtk file.
+    """
+
+    time_state = 0
+    while time_state < reference_length:
```

Comment:
Did we ever get a check in that all of the vector tags are the same length?

@gonuke we thought this was an okay assumption to start with and that support for variable length vector tags in the same file could come later.

Reply:
In the long term, I agree. I think *to start with* this is an ok assumption for this PR and that some further discussion is going to be required on how to write vector tags of different length. 

Several questions arise when we start to look to support that. How would VisIt handle viewing a data in a database where the tags are of different lengths and may appear in some files but not others? Does each vector tag then get its own VisIt database? These are questions I think we can answer down the line. This implementation gives us a very useful capability to build on.

Reply:
In this PR, `TagExpansion.py` supports the expansion of one vector tag per element in the mesh. I have opened #31 and I intend to handle multiple vector tags of different lengths in a future PR!